internal/privatemessaging/privatemessaging.go

```diff
@@ -93,6 +96,8 @@ func NewPrivateMessaging(ctx context.Context, di database.Plugin, im identity.Ma
 		},
 		opCorrelationRetries:  config.GetInt(config.PrivateMessagingOpCorrelationRetries),
 		maxBatchPayloadLength: config.GetByteSize(config.PrivateMessagingBatchPayloadLimit),
+		metricsEnabled:        config.GetBool(config.MetricsEnabled),
+		metrics:               mm,
```

Comment:
Maybe the metrics manager should be responsible for exposing the "are metrics enabled" flag, instead of each component reading it directly from the config?